lib/pacemaker/pcmk_acl.c

```diff
@@ -41,6 +41,19 @@ static const xmlChar *NS_WRITABLE = (const xmlChar *) ACL_NS_PREFIX "writable";
 static const xmlChar *NS_READABLE = (const xmlChar *) ACL_NS_PREFIX "readable";
 static const xmlChar *NS_DENIED =   (const xmlChar *) ACL_NS_PREFIX "denied";
 
+/*!
+ * \brief No clue
```

Comment:
Yeah we should probably rename the functions

This function takes some XML, and annotates it with XML namespaces to indicate the ACL permissions. 

It's interesting it handles XML_ELEMENT_NODE and XML_ATTRIBUTE_NODE. Element is obviously the main thing we're interested in, but I'm not sure why we'd annotate individual attributes since ACLs aren't that fine-grained. I'm actually not sure whether ACLs apply to comments (XML_COMMENT_NODE); that would be worth testing out.

Reply:
I tried handling XML comment nodes in `pcmk_acl.c`, but that change doesn't seem to make a difference; maybe `access-render-2.xsl` would need to be modified, too. So the help example now uses `| less -r`.

Reply:
That's fine -- that's what -r/-R is for, and it's a bonus that it works at all without it. I just forgot about -r when I initially tested it.